### PR TITLE
[JENKINS-35017] Migrate to new parent pom

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-35017](https://issues.jenkins-ci.org/browse/JENKINS-35017)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -54,7 +54,8 @@
   </licenses>
 
   <properties>
-    <maven.findbugs.failure.strict>true</maven.findbugs.failure.strict>
+    <jenkins.version>1.596.1</jenkins.version>
```
Comment (andresrc) [inline pom.xml]: Baseline bump motivated by `credentials-binding` dependency.

**Comment by armfergom** [comment]
Looks good. 🐝

File: pom.xml (hunk 1 above)
Comment (kwhetstone) [inline pom.xml]: This couldn't be bumped to 1.625?

**Comment by kwhetstone** [comment]
:bee:

**Comment by recena** [comment]
@andresrc Is this ready to merge?

File: pom.xml (hunk 1 above)
Comment (andresrc) [inline pom.xml]: The idea is keep the baselines at the older possible/reasonable version.

**Comment by andresrc** [comment]
@reviewbybees done

**Comment by andresrc** [comment]
@ndeloof ok to merge?

**Comment by ndeloof** [comment]
@andresrc I'm not a maintainer on this plugin, feel free to merge if you feel confident with your changes.

**Comment by andresrc** [comment]
@reviewbybees again

**Comment by armfergom** [comment]
🐝

**Comment by kwhetstone** [comment]
:bee: Looks like this can be merged.

**Pull request merged**